experiments/ClimaCore/sea-ice/slab_ice.jl

```diff
@@ -0,0 +1,341 @@
+const CI = !isnothing(get(ENV, "CI", nothing))
+
+# ## Define Parameters
+#  - Global Constants
+const FT = Float64;
+
+#  - Experiment-specific Parameters
+parameters = (
+    ## atmos parameters
+    zmin_atm = FT(0.0), # height of atm stack bottom [m] 
+    zmax_atm = FT(1.0), # height of atm stack top [m]
+    n = 15,  # number of elements in atm stack 
+    μ = FT(0.0001), # diffusion coefficient [m^2 / s]
+    T_top = FT(280.0), # fixed temperature at the top of the domain_atm [K]
+    ## slab ice parameters
+    ρc_ml = FT(4e6), # for mixed layer
+    F_basal = FT(120), # W / m2 / K
+    T_base = FT(273.16), # K 
+    L_ice = FT(3e8), # J / m3
+    h_ml = FT(1), # m 
+    T_freeze = FT(273.16), # K 
+    k_ice = FT(2), # W / m / K
+    ## coupling parameters 
+    λ = FT(1e6), # transfer coefficient 
+)
+
+# ## Define Model Functions
+
+# - Model 1 (atm) Equations
+"""
+    ∑tendencies_atm!(du, u, (parameters, T_sfc), t)
+
+Heat diffusion equation
+    dT/dt =  ∇ μ ∇ T
+    where
+        T  = 280 K              at z = zmax_atm
+        dT/dt = - ∇ F_sfc       at z = zmin_atm
+
+We also use this model to calculate and accumulate the downward surface fluxes, F_sfc:
+    F_sfc = - λ * (T_sfc - T1) 
+    d(F_integrated)/dt  = F_sfc
+    where
+        F_integrated is reset to 0 at the beginning of each coupling cycle
+        T1 = atm temperature near the surface (here assumed equal to the first model level)
+"""
+function ∑tendencies_atm!(du, u, p, t)
+
+    FT = eltype(u)
+    T = u.T # u.x = vector of prognostic variables from DifferentialEquations
+    F_sfc = calculate_flux(p.Ya.T_sfc[1], parent(T)[1], p.p)
+
+    ## set BCs
+    bcs_bottom = Operators.SetValue(Geometry.WVector(F_sfc ./ p.p.λ))
+    bcs_top = Operators.SetValue(Geometry.WVector(FT(0)))
+
+    gradc2f = Operators.GradientC2F()
+    gradf2c = Operators.DivergenceF2C(bottom = bcs_bottom, top = bcs_top) # Neumann BC (face-to-center)
+
+    ## tendency calculations
+    @. du.T = gradf2c(p.p.μ * gradc2f(T)) # dT/dt
+    du.F .= -F_sfc[1] # d(F_integrated)/dt
+
+end
+
+
+get_∂F_atm∂T_sfc(p) = -p.λ # TODO: in GCM this will be: 4 σ T_sfc^3 + ||uh|| ρ cp (C_H + L(q_sat1 - q_sat) * dT^(-1))
+
+# - Model 2 (ice) Equations
+"""
+    solve_ice!(dT_sfc, T_sfc, (parameters, F_accumulated), t)
+
+slab RHS with an implicit solve ice and explicit (forward Euler) solve for ocean
+
+"""
+function solve_ice!(integ, Δt)
+
+    Y = integ.u
+    Ya = integ.p.Ya
+    p = integ.p.p
+
+    FT = eltype(Y)
+    ocean_qflux = FT(0)
+
+    # prognostic
+    T_ml = Y.T_ml
+    T_sfc = Y.T_sfc
+    h_ice = Y.h_ice
+
+    # auxiliary
+    F_atm = Ya.F_atm
+
+    ∂F_atm∂T_sfc = get_∂F_atm∂T_sfc(p) # this will be passed from atmos/SF.jl
+
+    ΔT_ml = similar(F_atm)
+    Δh_ice = similar(F_atm)
+    F_ice = similar(F_atm)
+    ΔT_base = similar(F_atm)
+
+    # ice thickness and mixed layer temperature changes due to atmosphereic and ocean fluxes 
+    if h_ice[1] <= 0 # ice-free 
```

Comment:
Im not sure if this really matters, but Im not sure how often something is =0. You could say h_ice <eps(FT).

Do you see h_ice as negative ever?

Reply:
It shouldn't be negative and the `@. Δh_ice = -h_ice` should take care of this, but since it is not an integer I left it there in case there were some FP errors. I can invert the argument and change it to `h_ice[1] > 0` for ice covered and `else` otherwise?